.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,43 @@
+name: nightly
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # run at 12 AM UTC
+
+jobs:
+  nightly:
+    runs-on: ubuntu-latest
+    name: Build and push nightly
+    strategy:
+      matrix:
+        component:
+          - deps
+          - extractor
+          - gateway
+          - indexer
+          - tracker
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Dependencies (extractor)
+        run: cd extractor && npm install
+        if: matrix.component == 'extractor'
```

Comment:
You shouldn't need this step. Dependency installation happens within the docker container.